libbpf-rs/src/map.rs

```diff
@@ -731,3 +734,117 @@ impl<'a> Iterator for MapKeyIter<'a> {
         }
     }
 }
+/// Represent a wrapper `bpf_map_info` struct pointer
+/// Provide ability to retrive the details of a certain map with its fd only
+#[derive(Debug)]
+pub struct MapInfo {
+    info: bpf_map_info,
+}
+
+impl MapInfo {
+    /// Create a MapInfo struct from a fd.
+    pub fn new(fd: i32) -> Result<Self> {
+        let mut map_info = unsafe { std::mem::zeroed::<bpf_map_info>() };
+        let mut size = std::mem::size_of_val(&map_info) as u32;
+        let ret = unsafe {
+            bpf_obj_get_info_by_fd(
+                fd,
+                &mut map_info as *mut bpf_map_info as *mut c_void,
+                &mut size as *mut u32,
+            )
+        };
+        if ret < 0 {
+            return Err(Error::System(errno::errno()));
+        }
+        if size != std::mem::size_of_val(&map_info) as _ {
+            return Err(Error::Internal(format!(
+                "Unexpected info size: {} expected {}",
+                size,
+                std::mem::size_of_val(&map_info)
+            )));
+        }
+        Ok(Self { info: map_info })
+    }
+    /// Get the map type
+    pub fn map_type(&self) -> MapType {
+        match MapType::try_from(self.info.type_) {
+            Ok(t) => t,
+            Err(_) => MapType::Unknown,
+        }
+    }
+    /// Get the map id
+    pub fn id(&self) -> u32 {
+        self.info.id
+    }
```

Comment:
In my opinion we shouldn't be adding wrappers for each and every member of the C struct. Would it be sufficient to have another `as_libbpf_*` cast? Or just make the `info` part public. It's not like it's an implementation detail, really.

Reply:
I make `info` publicized now